test/test_av.py

```diff
@@ -0,0 +1,265 @@
+"""Test the artificial viscosity functions."""
```

Comment:
IMO, this is missing a test for proper handing of boundary conditions, particularly in view of the fact that we had trouble with that.

Reply:
A test has been added to make sure the AV fluxes returned from the fluid BC constructs match those which we intended for them to be.  We collectively share a lack of knowledge about what the BCs should actually be for this operator (i.e., when computing the 2nd part; `div(alpha*indicator*grad(Q))`).  For the first (`grad(Q)`) part, this operator now mercifully uses `grad(Q)` from the fluid computation, and is thus using the appropriate fluid BCs directly there - whatever those may be for the given boundary.    The added tests check the AV-specific interface fluxes for the 2nd part.